File: smash-ranks-android/app/src/main/java/com/garpr/android/misc/ResultCodes.java
package com.garpr.android.misc;

import android.os.Parcel;
import android.os.Parcelable;

public enum ResultCodes implements Parcelable {

    IDENTITY_SELECTED,
    REGION_SELECTED,
    RINGTONE_SELECTED;

    public final int mValue;


    ResultCodes() {
        mValue = ordinal() + 1000;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(final Parcel dest, final int flags) {
        dest.writeInt(ordinal());
    }

    public static final Creator<ResultCodes> CREATOR = new Creator<ResultCodes>() {
        @Override
        public ResultCodes createFromParcel(final Parcel source) {
            return values()[source.readInt()];
        }

        @Override
        public ResultCodes[] newArray(final int size) {
            return new ResultCodes[size];
        }
    };

}

File: smash-ranks-android/app/src/main/java/com/garpr/android/misc/ResultCodes.kt
package com.garpr.android.misc

import android.os.Parcel
import android.os.Parcelable
import com.garpr.android.extensions.createParcel

enum class ResultCodes : Parcelable {

    IDENTITY_SELECTED,
    REGION_SELECTED,
    RINGTONE_SELECTED;


    companion object {
        @JvmField
        val CREATOR = createParcel { values()[it.readInt()] }
    }

    val value = ordinal + 1000

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(ordinal)
    }

}